src/convert/impls.rs

```diff
@@ -418,3 +418,16 @@ impl<T: IntoWasmAbi> ReturnWasmAbi for Result<T, JsValue> {
         }
     }
 }
+
+if_std! {
+    use std::string::ToString;
+
+    impl<T: IntoWasmAbi, E: std::error::Error> ReturnWasmAbi for Result<T, E> {
+        type Abi = T::Abi;
+
+        fn return_abi(self, extra: &mut Stack) -> Self::Abi {
+            self.map_err(|err| JsValue::error(&err.to_string()))
```

Comment:
Would it be possible to include more than `.to_string()` here? Sometimes an `Error` could be a stack of errors (following `.cause()`) which makes the most sense when presented all at once. Using just `Error` can we do some iteration here or something like that to build a larger message?

Reply:
We could, but I'm not sure how well that would interact with JS-generated stacktrace... Don't you think it might get a bit confusing to see two different but overlapping stacktraces on one error?

Reply:
Oh right I don't think it'd interact with the JS stack trace, only that we'd use the Rust stack trace to generate a more descriptive message

Reply:
Yeah but what I mean, when you throw it on the JS side, you'll get Rust stack trace in the message followed by JS stack trace right below (as that's what JS engines normally print right after the message). Hence my concern that it might be somewhat confusing to see both following each other when in fact parts of them overlap in terms of functions... Not sure how to address this nicely.

Reply:
It's true yeah it could get a bit garbled, but I actually think that's sort of what we want exactly, a Rust stack trace followed by a JS one!

We could perhaps experiment with various formattings though to see which works out best for the Rust message?

Reply:
I'm still not sure how I feel about multiline error message on JS side (it is rather unusual) and also about potential duplication of information within it (since Display implementations often tend to forward printing to nested errors, so adding `.source()` chain would produce duplicates of same messages already included in the first line).

While I totally agree that experimentation with different formats is useful, as well as with API itself, I still feel that it's not as simple as it seems at first glance and would prefer to leave such experiments to separate PRs.

Reply:
I would prefer to not land this unless it conveys the backtrace information for the Rust error. That often contains critical information about where errors come from needed for debugging, and as a built-in wrapper we in this crate would want to be sure to convey that information.

Reply:
Sorry, as mentioned, I'm not sure how this should look like (especially as I never used / looked into the `cause` chains in Rust land, usually relying just on stack backtraces) and on the other hand never seen multiline error messages in JS land. Taking these two experiences into account, I'm not really confident with adding this part myself.

Feel free to close the PR, I'm happy for someone else to take over and come up with a better solution.